src/olympia/addons/models.py

```diff
@@ -541,6 +548,35 @@ def from_upload(cls, upload, platforms, source=None, is_listed=True,
 
         return addon
 
+    @classmethod
+    def resolve_webext_translations_and_save(cls, addon, data, upload):
+        """Resolve all possible translations from an add-on."""
+        fields = ('name', 'homepage', 'summary')
+        messages = extract_translations(upload)
+
+        # This is a bit stupid but I don't see any other way.
+        # We cannot attach actual translations while the add-on isn't saved
```

Comment:
yeah, I'll take some more time and see what was going on here. I think maybe `resolve_translations` could mutate `data` before it's being set on the add-on instance - this way we avoid any weirdness. Guess I'll do that